### PR TITLE
Add redirects to developer site

### PR DESCRIPTION
**Pull request by technosophos**
This adds redirects from older spin docs to the developer site.

All I did was add a bunch of entries to a redirector, much like what we do on fermyon.com. The redirector is the same one from fermyon.com and Finicky Whiskers.

It removes all of the old content.

Signed-off-by: Matt Butcher <matt.butcher@fermyon.com>

### CODE REVIEW

**Comment by technosophos** [comment]
I also updated index.md so that anyone who comes into this directory looking for info will find an explanation of where to go instead.

File: docs/content/index.md
Comment (mikkelhegn) [inline docs/content/index.md:9]: Should we just redirect this as well - or do something like "Redirecting you in 3..2..1.."?

File: docs/spin.toml
Comment (mikkelhegn) [inline docs/spin.toml:8]: The download is just the install script, which we are also moving, so we can remove this as well, unless we cannot redirect those requests.

File: docs/spin.toml
Comment (mikkelhegn) [inline docs/spin.toml:153]: Is it intentional to not redirect to the specific SIP? Also for the following redirects.

File: docs/spin.toml
Comment (mikkelhegn) [inline docs/spin.toml:193]: This is still in GitHub

**Review by mikkelhegn — APPROVED** [review]
A few things to change, but approving now. We should not merge until the new site is ready. I can be on point for that.

File: docs/content/index.md
Comment (technosophos) [reply docs/content/index.md:9]: It is actually redirected already. I left this here mainly for people who are used to seeing the docs in GitHub so that they know where to go.

File: docs/spin.toml
Comment (technosophos) [reply docs/spin.toml:193]: Oops... missed that one. Will fix

File: docs/spin.toml
Comment (technosophos) [reply docs/spin.toml:153]: I can go either way on this. I wasn't sure what the best practice was when linking back to GitHub. So I just linked them all to the main SIP index.

File: docs/spin.toml
Comment (technosophos) [reply docs/spin.toml:8]: Fixed

File: docs/spin.toml
Comment (technosophos) [reply docs/spin.toml]: Fixed

File: docs/spin.toml
Comment (technosophos) [reply docs/spin.toml]: Fixed

File: docs/content/index.md
Comment (mikkelhegn) [reply docs/content/index.md:9]: Good point!

**Comment by lann** [comment]
This seems to have broken one of our CI checks: https://github.com/fermyon/spin/actions/runs/3300082364/jobs/5444179172

**Comment by lann** [comment]
aaand its already fixed https://github.com/fermyon/spin/pull/846

**Pull request merged**
